gravitee-apim-gateway/gravitee-apim-gateway-reactor/src/main/java/io/gravitee/gateway/reactor/handler/AbstractAccessPointAcceptor.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.gateway.reactor.handler;
+
+import static java.util.stream.Collectors.toCollection;
+
+import io.gravitee.common.event.Event;
+import io.gravitee.common.event.EventListener;
+import io.gravitee.common.event.EventManager;
+import io.gravitee.gateway.reactor.accesspoint.AccessPointEvent;
+import io.gravitee.gateway.reactor.accesspoint.ReactableAccessPoint;
+import jakarta.validation.constraints.NotNull;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+/**
+ * @author GraviteeSource Team
+ */
+public abstract class AbstractAccessPointAcceptor<T extends Acceptor<T>>
+    implements Acceptor<T>, EventListener<AccessPointEvent, ReactableAccessPoint> {
+
+    protected final ReactorHandler apiReactor;
+    private final String environmentId;
+    private final EventManager eventManager;
+    protected List<T> acceptors;
+
+    public AbstractAccessPointAcceptor(ReactorHandler apiReactor, final EventManager eventManager, final String environmentId) {
+        this.apiReactor = apiReactor;
+        this.environmentId = environmentId;
+        this.eventManager = eventManager;
+        this.eventManager.subscribeForEvents(this, AccessPointEvent.class);
+    }
+
+    protected void initializeAcceptors(final List<ReactableAccessPoint> initialReactableAccessPoints) {
+        this.acceptors =
+            initialReactableAccessPoints
+                .stream()
+                .filter(reactableAccessPoint -> reactableAccessPoint.getTarget() == getTarget())
+                .map((ReactableAccessPoint reactableAccessPoint) -> createAcceptor(reactableAccessPoint.getHost()))
+                .collect(toCollection(CopyOnWriteArrayList::new));
+    }
+
+    protected abstract T createAcceptor(String accessPointHost);
+
+    protected abstract ReactableAccessPoint.Target getTarget();
+
+    public abstract String toString();
```

Comment:
I don't know if toString is the best naming, but since we need some sort of uuid for compareTo, I thought it might do the trick. However, I'm not sure what the compareTo is for

Reply:
See my next comment